### DISCUSSION TITLE
How to determine if we are tailing at the moment?

### DISCUSSION THREAD

**Discussion by gajus** [petyosi/react-virtuoso#498] [Q&A] [answered]
There is the `followOutput` – however, how do I tell if we are currently tailing or not?

**Comment by petyosi** [comment, marked as answer]
[`atBottomStateChange`](https://virtuoso.dev/virtuoso-api-reference/#virtuoso-properties).

**Reply by gajus** [reply]
`atBottomStateChange` didn't work for my use case (since I needed to check if we are at the bottom now outside of the callback), but this worked:

```ts
const [visibleRange, setVisibleRange] = useState({
  endIndex: 0,
  startIndex: 0,
});

console.log(filteredMessages.length === visibleRange.endIndex + 1);

<Virtuoso
  data={filteredMessages}
  followOutput
  rangeChanged={setVisibleRange}
/>

```

**Reply by gajus** [reply]
Digging deeper this made me realize that the way we were using `Virtuoso` was not memoized properly, i.e. `followOutput` was not even working as intended.

Removing `data` in favor of `itemContent={itemContent}` and `totalCount={filteredMessageCount}` removed the need for needing to know if we are following output because whatever we set `followOutput` to accurately reflects if we are following or not.
